tools/update-codemirror-themes.js

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env node
+'use strict';
+
+const fs = require('fs-extra');
+const path = require('path');
+
+// Update theme names list in codemirror-editing-hook.js
+async function getThemes() {
+  const p = path.join(__dirname, '..', 'vendor/codemirror/theme/');
+  const files = await fs.readdir(p);
+  return files
+    .filter(name => name.endsWith('.css'))
+    .map(name => name.replace('.css', ''))
+    .sort();
+}
+
+function replaceThemes(content, themes) {
+  const list = JSON.stringify(themes, null, 8).replace(/"/g, '\'');
+  return content.replace(
+    /\/\*\s*populate-theme-start\s*\*\/[\s\S]+\/\*\s*populate-theme-end\s*\*\//,
+    // strip off square brackets & first 8 spaces
```

Comment:
Hardcoding the number of spaces and the type of linefeed is meh.
It's better to extract it from the current file.

```js
function replaceThemes(content, themes) {
  const lineFeed = content.indexOf('\r\n') ? '\r\n' : '\n';
  return content.replace(
    /(\x20+)(\/\*\s*populate-theme-start\s*\*\/)[\s\S]+?(\/\*\s*populate-theme-end\s*\*\/)/,
    (_, indent, intro, outro) =>
      indent + intro + lineFeed +
      themes.map(_ => `${indent}'${_}'`).join(lineFeed) + lineFeed +
      outro
  );
}
```

Not tested.

Reply:
looks good.. I'll add this. The only thing missing was an `indent + outro` instead of just `outro`.